programmers/난이도별/level01.K번째수/HyeonJeong.py

```diff
@@ -0,0 +1,43 @@
+# 1. 첫 시도
+'''
+def solution(array, commands):
+    answer = []
+    for j in range(len(commands)) :
+        new_array = []
+        for i in range(commands[j][0] - 1, commands[j][1]) :
+            new_array.append(array [i])
+        new_array.sort()
+        answer.append(new_array [(commands [j][2] - 1)])
+    return answer
+'''
+
+# 2. list[:]이용
+'''
+def solution(array, commands):
+    answer = []
+    for j in range(len(commands)) :
+        new_array =  array [commands[j][0] - 1 : commands[j][1]]
+        new_array.sort()
+        answer.append(new_array [(commands [j][2] - 1)])
+    return answer
+'''
+
+# 3. sorted()이용
+'''
+def solution(array, commands):
+    answer = []
+    for j in range(len(commands)) :
+        new_array =  sorted(array [commands[j][0] - 1 : commands[j][1]])
+        answer.append(new_array [(commands [j][2] - 1)])
+    return answer
+'''
+
+# 4. 더 짧은 코드 이용
+def solution(array, commands):
+    answer = []
+    for j in range(len(commands)) :
+        answer.append(sorted(array [commands[j][0] - 1 : commands[j][1]])[(commands [j][2] - 1)])
+    return answer
```

Comment:
```suggestion
def solution(array, commands):
    answer = []
    for command in commands :
        x, y, z = command
        answer.append(sorted(array[x - 1:y])[(z - 1)])
    return answer
```
이런식으로 표현할 수 있고 혹은 아예 한 줄로 표현할 수 있습니다.

```python
def solution(array, commands):
    return [ sorted(array[command[0] - 1 : command[1]])[command[2] - 1]  for command in commands ]
```

```python
def solution(array, commands):
    return [ sorted( array[v[0] - 1 : v[1]] )[v[2] - 1]  for vin commands ]
```

Reply:
x, y, z = command 사용이 가능하다는 것을 처음으로 알게 되었네요!
가독성이 좋아지는 코드를  알려 주셔서 감사합니다.